### PR TITLE
trying to move eslint to local again

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers This has to pass CI so... we'll see in a few.
Also you'll have to setup your dev environments to load local eslint and use that correctly. I'd uninstall global eslint ASAP else you'll get incorrect linting.

Also this updates the pre-commit hook so run ```pre-commit install```

I'm using eslint-cli (npm -g eslint-cli) which dynamically points your eslint to whatever node_modules eslint you're in (works in vim)

### CODE REVIEW

File: desktop/npm-helper.js [hunk 1]
```diff
@@ -3,21 +3,6 @@ import path from 'path'
 import childProcess, {execSync} from 'child_process'
 import fs from 'fs'
 
-const postinstallGlobals = {
```
Comment (chrisnojima) [inline desktop/npm-helper.js]: this moved into our package.json

File: desktop/npm-shrinkwrap.json [hunk 2]
```diff
@@ -576,9 +596,9 @@
...
     },
     "babel-types": {
-      "version": "6.9.1",
+      "version": "6.10.0",
       "from": "babel-types@>=6.9.1 <7.0.0",
-      "resolved": "https://registry.npmjs.org/babel-types/-/babel-types-6.9.1.tgz"
+      "resolved": "https://registry.npmjs.org/babel-types/-/babel-types-6.10.0.tgz"
     },
     "babylon": {
       "version": "6.8.0",
```
Comment (chrisnojima) [inline desktop/npm-shrinkwrap.json]: this is manually edited to be 6.8.0 which is the last known working version

**Comment by MarcoPolo** [comment]
👍 

I wish changing a dep in the package.json didn't lead to a huge change in the shrinkwrap file :(

**Comment by chrisnojima** [comment]
well theoretically we could try and limit the tree but... what a pain

**Comment by awendland** [comment]
👍 

Works for my dev-flow once I use this `npm bin` alias setup @ http://stackoverflow.com/a/15157360

**Comment by MarcoPolo** [comment]
works for me on neovim and neomake

**Comment by cjb** [comment]
Doesn't work with Atom, but I can handle maintaining my own global eslint.

:thumbsup:

**Comment by chrisnojima** [comment]
@cjb you'll update the js tests thing right? merging now

**Pull request merged**